lib/internal/errors.js

```diff
@@ -574,6 +574,22 @@ function oneOf(expected, thing) {
   }
 }
 
+function addNumericalSeparator(val) {
+  let res = val[val.length - 1];
+  let count = 0;
+  for (let i = val.length - 2; i >= 0; i--) {
+    count++;
+    if (val[i] === '.') {
+      return val;
+    }
+    if (count % 3 === 0 && i !== val.length && val[i] !== '-') {
```

Comment:
`i` can never be `val.length` in the body of this `for` loop.

Reply:
Oh, that's a remnant of a former version. Thanks!